shared/node.go

```diff
@@ -28,32 +29,38 @@ type Node struct {
 }
 
 func Create(configPath, genesisPath string) (n *Node, err error) {
-	persistenceMod, err := persistence.Create(configPath, genesisPath)
+
+	cfg, genesis, err := runtime.Init(configPath, genesisPath)
+	if err != nil {
+		return nil, err
+	}
+
+	persistenceMod, err := persistence.Create(cfg.Persistence, genesis.PersistenceGenesisState)
 	if err != nil {
 		return nil, err
 	}
 
-	p2pMod, err := p2p.Create(configPath, genesisPath, false)
+	p2pMod, err := p2p.Create(cfg.P2P, false)
```

Comment:
The design pattern for module creation is loosened here

Reply:
Maybe this is better
![image](https://user-images.githubusercontent.com/29378614/191291630-bf413a2f-1e86-48cb-8a6a-0d1a3303e1ad.png)

maybe builder is not the best naming here

![image](https://user-images.githubusercontent.com/29378614/191291874-93df89ed-8686-4ae5-afa2-c6846a8cf981.png)

![image](https://user-images.githubusercontent.com/29378614/191291920-31cb9ba5-8188-47a4-8cae-5e47c5dcb9ad.png)

![image](https://user-images.githubusercontent.com/29378614/191292206-455e3163-ae51-4c77-bf1f-d13f6d64057c.png)

Reply:
Yes, the `getRandomPK` should go away soon, but I like this a log better if you guys do

++ I renaming NewBuilder though